errors.go

```diff
@@ -84,6 +85,10 @@ func IsInvalidFieldNameError(err error) bool {
 	return hasError(err, ErrInvalidFieldName)
 }
 
+func IsConditionViolatedError(err error) bool {
```

Comment:
looks good.  would you mind changing the name to match the error e.g. IsConditionalCheckFailedException and then I think we're good to go

Reply:
Done! Added a basic test as well